### DISCUSSION TITLE
remove `/` before `publicPath` in .mdoc file?

### DISCUSSION THREAD

**Discussion by radenpioneer** [Thinkmill/keystatic#330] [Q&A] [answered]
I am trying to use Keystatic along with Astro Collections and Astro's (still experimental) assets feature.

In my Keystatic config:

```ts
collections: {
    blog: collection({
     // ... other collection config
      format: { contentField: 'content' },
      schema: {
        // ... other schema config
        content: fields.document({
         // ... other document config
          images: {
            directory: 'src/assets/uploads/blog',
            publicPath: '../../assets/uploads/blog/',
          },
        }),
      },
    }),
}
```

This is the result of above configuration:
```mdoc
---
title: A Title
---

This is first line
This is second line

![](/../../assets/uploads/blog/image.jpg)
```

The image's `publicPath` automatically appended after `/`, resulting the image markup using absolute path, while I wanted the image to use relative path (`![](../../assets/uploads/blog/image.jpg)`) instead of absolute path so it may take advantage of Astro's assets optimization feature.

Any way to change this?

**Comment by eyrewiut** [comment]
I have run into this problem as well, @radenpioneer did you find a solution?

**Reply by radenpioneer** [reply]
I didn't. I ended up moving images to `/public` instead.

**Reply by eyrewiut** [reply]
I see, thats a shame. In case its of any help, I just tried switching to the [astro-imagetools](https://astro-imagetools-docs.vercel.app/en/introduction) integration and it works extremely well. I think I will try and get it working with that instead.. 
Unfortunately the build fails because astro-imagetools doesn't have usable ssr support 😓

**Comment by simonswiss** [comment, marked as answer]
Hey!

**Update**: It's now possible to do this — see comment below 👀

~Unfortunately it's not possible right now~, but I'll post an update here if this changes in the future 👍

**Reply by simonswiss** [reply]
Update: This is now possible — see [this comment](https://github.com/Thinkmill/keystatic/discussions/372#discussioncomment-6484749) for context.

**Reply by radenpioneer** [reply]
Thanks, @simonswiss!

**Reply by simonswiss** [reply]
No worries — hopefully that should unblock you to do a lot of things with Astro. It did for me! 😃
